packages/realm-react/src/useQuery.tsx

```diff
@@ -44,10 +44,14 @@ export function createUseQuery(useRealm: () => Realm) {
   function useQuery<T extends Realm.Object<any>>(
     type: string | RealmClassType<T>,
     query: QueryCallback<T> = (collection: Realm.Results<T>) => collection,
-    deps: DependencyList = [],
+    deps: DependencyList = [type],
```

Comment:
With the introduction of `requiredDeps` this could continue defaulting to `[]`, otherwise this would be duplicated in the final `deps`.

Reply:
Good catch, I forgot to revert this.